nautobot/core/testing/api.py

```diff
@@ -353,24 +353,27 @@ def test_create_object_without_permission(self):
                 self.assertHttpStatus(response, status.HTTP_403_FORBIDDEN)
 
         def check_expected_slug(self, obj):
-            slug_source = self.slug_source if isinstance(self.slug_source, (list, tuple)) else [self.slug_source]
-            expected_slug = ""
-            for source_item in slug_source:
-                # e.g. self.slug_source = ["parent__name", "name"]
-                source_keys = source_item.split("__")
-                try:
-                    val = getattr(obj, source_keys[0])
-                    for key in source_keys[1:]:
-                        val = getattr(val, key)
-                except AttributeError:
-                    val = ""
-                if val:
-                    if expected_slug != "":
-                        expected_slug += "-"
-                    expected_slug += self.slugify_function(val)
-
-            self.assertNotEqual(expected_slug, "")
-            self.assertEqual(obj.slug, expected_slug)
+            # slug field has been removed from alot of models,
+            # causing this test to fail for such models
+            if hasattr(obj, "slug"):
```

Comment:
I'd think it would make more sense to check whether `self.slug_source` is defined as a test parameter, and skip the test if it's not? That way the test controls whether slug is tested or not, instead of the model controlling it.